### ISSUE TITLE
DFU device checking not foolproof

### ISSUE THREAD

**Issue by kennethlimcp** [spark/particle-cli#174]
From the forums:

Somehow CLI tries to flash to a non-Particle device

```
And here's the output for particle flash --usb electron_firmware_1448459332199.bin when the bluetooth device is enabled.

Cannot open DFU device 0a5c:217f
Cannot open DFU device 0a5c:217f

Error writing firmware...Cannot open DFU device 0a5c:217f
Cannot open DFU device 0a5c:217f
```

**Comment by dmiddlecamp** [comment]
Thanks for opening this one, I've definitely seen reports of this as well.  I wasn't sure if this was a bug in dfu-util, or the CLI, didn't we usually pipe out the DFU command before we run it?

**Comment by kennethlimcp** [comment]
We concentanate our own DFU command but the matching is not saying "no devices" found and went ahead to execute the DFU command

**Comment by brycekahle** [comment]
I cannot reproduce this, checking with the forum user for CLI version information.

**Comment by brycekahle** [comment]
I fixed this in a5a75ab2018d127ccf0ae2ef62fd34aaa04837a1
